File: backend/src/main/java/com/culinaryheaven/domain/user/dto/response/FollowResponse.java
package com.culinaryheaven.domain.user.dto.response;

import com.culinaryheaven.domain.user.domain.Follow;
import io.swagger.v3.oas.annotations.media.Schema;

public record FollowResponse(

        @Schema(description = "팔로우한 유저 ID")
        Long userId,

        @Schema(description = "팔로우한 유저 닉네임")
        String nickname

) {
        public static FollowResponse of(Follow follow) {
                return new FollowResponse(follow.getTarget().getId(), follow.getTarget().getUsername());
        }
}

File: android/data/src/main/java/com/jae464/data/remote/model/response/user/FollowResponse.kt
package com.jae464.data.remote.model.response.user

import com.jae464.domain.model.Follow
import kotlinx.serialization.Serializable

@Serializable
data class FollowResponse(
    val userId: Long,
    val nickname: String
)

fun FollowResponse.toDomain(): Follow {
    return Follow(userId = userId, nickname = nickname)
}
